os/arch/arm/src/amebasmart/amebasmart_reboot_reason.c

```diff
@@ -118,6 +118,15 @@ static reboot_reason_code_t up_reboot_reason_get_hw_value(void)
 
 		/* CA32:AP or KM4:NP or KM0:LP System reset */
 		else if ((boot_reason & AON_BIT_RSTF_APSYS) || (boot_reason & AON_BIT_RSTF_NPSYS) || (boot_reason & AON_BIT_RSTF_LPSYS)) {
+			if (boot_reason & AON_BIT_RSTF_APSYS) {		/* CA32 */
+				lldbg("Reboot reason: APSYS reset\n");
+			}
+			if (boot_reason & AON_BIT_RSTF_NPSYS) {		/* KM4 */
+				lldbg("Reboot reason: NPSYS reset\n");
+			}
+			if (boot_reason & AON_BIT_RSTF_LPSYS) {		/* KM0 */
+				lldbg("Reboot reason: LPSYS reset\n");
+			}
```

Comment:
Is it possible to be several cases at the same time? not exclusive?
I mean
if
else if
else if
is not correct?